src/ImageSharp/Processing/Processors/Transforms/Automorphic/AffineTransformProcessor{TPixel}.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System.Numerics;
+using SixLabors.ImageSharp.PixelFormats;
+
+namespace SixLabors.ImageSharp.Processing.Processors.Transforms
+{
+    /// <summary>
+    /// Provides the base methods to perform affine transforms on an image.
+    /// </summary>
+    /// <typeparam name="TPixel">The pixel format.</typeparam>
+    internal partial class AffineTransformProcessor<TPixel> : TransformProcessor<TPixel>, IResamplingImageProcessor<TPixel>
+        where TPixel : struct, IPixel<TPixel>
+    {
+        private readonly Size destinationSize;
+        private readonly Matrix3x2 transformMatrix;
+        private readonly IResampler resampler;
+        private ImageFrame<TPixel> source;
+        private ImageFrame<TPixel> destination;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AffineTransformProcessor{TPixel}"/> class.
+        /// </summary>
+        /// <param name="configuration">The configuration which allows altering default behaviour or extending the library.</param>
+        /// <param name="definition">The <see cref="AffineTransformProcessor"/> defining the processor parameters.</param>
+        /// <param name="source">The source <see cref="Image{TPixel}"/> for the current processor instance.</param>
+        /// <param name="sourceRectangle">The source area to process for the current processor instance.</param>
+        public AffineTransformProcessor(Configuration configuration, AffineTransformProcessor definition, Image<TPixel> source, Rectangle sourceRectangle)
+            : base(configuration, source, sourceRectangle)
+        {
+            this.destinationSize = definition.DestinationSize;
+            this.transformMatrix = definition.TransformMatrix;
+            this.resampler = definition.Sampler;
+        }
+
+        protected override Size GetDestinationSize() => this.destinationSize;
+
+        /// <inheritdoc/>
+        protected override void OnFrameApply(ImageFrame<TPixel> source, ImageFrame<TPixel> destination)
+        {
+            this.source = source;
+            this.destination = destination;
+            this.resampler.ApplyTransform(this);
+        }
+
+        /// <inheritdoc/>
+        public void ApplyTransform<TResampler>(in TResampler sampler)
+            where TResampler : struct, IResampler
+            => ApplyAffineTransform(
```

Comment:
nit: wouldn't it be simpler to just inline the contents of the method here? I don't see many benefits from the separate `.Transforms.cs` files either.

Reply:
Yeah, was just being lazy and didn't want to bog the classes down with too much code. Will update.

Reply:
Yeah, was just being lazy and didn't want to bog the classes down with too much code. Will update.